crates/bevy_animation/src/lib.rs

```diff
@@ -968,70 +929,117 @@ pub fn animate_targets(
                 return;
             };
 
+            let Some(threaded_animation_graph) =
+                threaded_animation_graphs.0.get(&animation_graph_id)
+            else {
+                return;
+            };
+
             // Determine which mask groups this animation target belongs to.
             let target_mask = animation_graph
                 .mask_groups
                 .get(&target_id)
                 .cloned()
                 .unwrap_or_default();
 
-            // Apply the animations one after another. The way we accumulate
-            // weights ensures that the order we apply them in doesn't matter.
-            //
-            // Proof: Consider three animations A₀, A₁, A₂, … with weights w₀,
-            // w₁, w₂, … respectively. We seek the value:
-            //
-            //     A₀w₀ + A₁w₁ + A₂w₂ + ⋯
-            //
-            // Defining lerp(a, b, t) = a + t(b - a), we have:
-            //
-            //                                    ⎛    ⎛          w₁   ⎞           w₂     ⎞
-            //     A₀w₀ + A₁w₁ + A₂w₂ + ⋯ = ⋯ lerp⎜lerp⎜A₀, A₁, ⎯⎯⎯⎯⎯⎯⎯⎯⎟, A₂, ⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎟ ⋯
-            //                                    ⎝    ⎝        w₀ + w₁⎠      w₀ + w₁ + w₂⎠
-            //
-            // Each step of the following loop corresponds to one of the lerp
-            // operations above.
-            let mut total_weight = 0.0;
-            for (&animation_graph_node_index, active_animation) in
-                animation_player.active_animations.iter()
-            {
-                // If the weight is zero or the current animation target is
-                // masked out, stop here.
-                if active_animation.weight == 0.0
-                    || (target_mask & active_animation.computed_mask) != 0
-                {
-                    continue;
-                }
+            let mut evaluation_state = animation_evaluation_state.get_or_default().borrow_mut();
+            let evaluation_state = &mut *evaluation_state;
 
-                let Some(clip) = animation_graph
-                    .get(animation_graph_node_index)
-                    .and_then(|animation_graph_node| animation_graph_node.clip.as_ref())
-                    .and_then(|animation_clip_handle| clips.get(animation_clip_handle))
+            // Evaluate the graph.
+            for &animation_graph_node_index in threaded_animation_graph.threaded_graph.iter() {
+                let Some(animation_graph_node) = animation_graph.get(animation_graph_node_index)
                 else {
                     continue;
                 };
 
-                let Some(curves) = clip.curves_for_target(target_id) else {
-                    continue;
-                };
+                match animation_graph_node.clip {
+                    None => {
+                        // This is a blend node.
```

Comment:
This isn't actually true because if all the weights we've seen thus far are zero then the `AnimationCurveEvaluator` won't have been created yet. However, we correctly handle this case by doing nothing.

Reply:
Cool! That's good to know.